src/software/jetson_nano/services/motor.cpp

```diff
@@ -119,13 +130,38 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
 MotorService::~MotorService() {}
 
 std::unique_ptr<TbotsProto::DriveUnitStatus> MotorService::poll(
-    const TbotsProto::DirectControlPrimitive_DirectVelocityControl& local_velocity,
-    float dribbler_speed_rpm)
+    const TbotsProto::DirectControlPrimitive& direct_control)
 {
-    // TODO (#2335) convert local velocity to per-wheel velocity
-    // using http://robocup.mi.fu-berlin.de/buch/omnidrive.pdf and then
-    // communicate velocities to trinamic. Also read back feedback and
-    // return drive unit status.
+    CHECK(encoder_calibrated_[FRONT_LEFT_MOTOR_CHIP_SELECT])
+        << "Running without encoder calibration can cause serious harm";
+
+    switch (direct_control.wheel_control_case())
+    {
+        case TbotsProto::DirectControlPrimitive::WheelControlCase::kDirectPerWheelControl:
+        {
+            tmc4671_setTargetVelocity(
+                FRONT_LEFT_MOTOR_CHIP_SELECT,
+                static_cast<int>(
+                    direct_control.direct_per_wheel_control().front_left_wheel_rpm() *
+                    MAX_DRIVE_RPM));
```

Comment:
why is the target velocity set to units of RPM^2?

Reply:
bad naming thing, its effectively a random scaling factor, so I changed it to `RANDOM_SCALING_FACTOR` 

I made a ticket to update it considering the gear ratio (or figuring out why rpm in robot diag is not rpm on trinamic)